### PR TITLE
feat: Get ERC-1155 token name from contractURI getter fallback

### PR DESCRIPTION
**Pull request by vbaranov**
A part of https://github.com/blockscout/blockscout/issues/10145

## Motivation

Empty names of most of ERC_1155 tokens.

## Changelog

Add a fallback to `contractURI` getter method for ERC-1155 tokens.

## Checklist for your Pull Request (PR)

  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from silently reappearing again.
  - [x] I checked whether I should update the docs and did so by submitting a PR to https://github.com/blockscout/docs
  - [x] If I added/changed/removed ENV var, I submitted a PR to https://github.com/blockscout/docs to update the list of env vars at https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md and I updated the version to `master` in the Version column. Changes will be reflected in this table: https://docs.blockscout.com/for-developers/information-and-settings/env-variables. 
  - [x] If I added new DB indices, I checked, that they are not redundant with PGHero or other tools.
  - [x] If I added/removed chain type, I modified the Github CI matrix and PR labels accordingly.

### CODE REVIEW

**Comment by drstrangelove111** [comment]
👀

File: apps/explorer/mix.exs [hunk 1]
```diff
@@ -140,7 +140,7 @@ defmodule Explorer.Mixfile do
   defp env_aliases(:dev), do: []
 
   defp env_aliases(_env) do
-    [compile: "compile --warnings-as-errors"]
```
Comment (sl1depengwyn) [inline apps/explorer/mix.exs:143]: What is the reason for this?

File: apps/explorer/lib/explorer/token/metadata_retriever.ex [hunk 2]
```diff
@@ -84,33 +98,62 @@ defmodule Explorer.Token.MetadataRetriever do
...
   @contract_functions %{
-    "18160ddd" => [],
-    "313ce567" => [],
-    "06fdde03" => [],
-    "95d89b41" => []
+    @name_signature => [],
+    @decimals_signature => [],
+    @name_signature => [],
+    @symbol_signature => [],
+    @total_supply_signature => []
```
Comment (fedor-ivn) [inline apps/explorer/lib/explorer/token/metadata_retriever.ex:130]: Could there be any use case in the future when arguments are not empty (but hardcoded).

File: apps/explorer/lib/explorer/chain/token.ex [hunk 3]
```diff
@@ -177,7 +177,7 @@ defmodule Explorer.Chain.Token do
 
     from(
       token in __MODULE__,
-      select: token.contract_address_hash,
+      select: token,
```
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/explorer/lib/explorer/chain/token.ex]: Processed in https://github.com/blockscout/blockscout/pull/10187/commits/86c139ce08d357ee958f122c909afee87e3157c0

File: apps/explorer/lib/explorer/token/metadata_retriever.ex [hunk 4]
```diff
@@ -84,33 +98,62 @@ defmodule Explorer.Token.MetadataRetriever do
...
   # 06fdde03 = keccak256(name())
+  @name_signature "06fdde03"
   # 95d89b41 = keccak256(symbol())
+  @symbol_signature "95d89b41"
   @contract_functions %{
-    "18160ddd" => [],
-    "313ce567" => [],
-    "06fdde03" => [],
-    "95d89b41" => []
+    @name_signature => [],
```
[the remark this answers is not in the document]
Comment (vbaranov) [reply apps/explorer/lib/explorer/token/metadata_retriever.ex]: Processed in https://github.com/blockscout/blockscout/pull/10187/commits/86c139ce08d357ee958f122c909afee87e3157c0

File: apps/explorer/mix.exs (hunk 1 above)
Comment (vbaranov) [reply apps/explorer/mix.exs]: Ouch, this is a testing artefact. Removed in https://github.com/blockscout/blockscout/pull/10187/commits/86c139ce08d357ee958f122c909afee87e3157c0.

File: apps/explorer/lib/explorer/token/metadata_retriever.ex (hunk 2 above)
Comment (vbaranov) [reply apps/explorer/lib/explorer/token/metadata_retriever.ex:129]: Theoretically, this is possible.

**Pull request merged**